websocket_tls/nbioclient_ordered_messages/client.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/url"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/lesismal/nbio/nbhttp"
+	"github.com/lesismal/nbio/nbhttp/websocket"
+
+	"github.com/lesismal/llib/std/crypto/tls"
+)
+
+var (
+	clients           = flag.Int("clients", 1, "number of clients")
+	connectedClients  chan *websocket.Conn
+)
+
+func newUpgrader() *websocket.Upgrader {
+	u := websocket.NewUpgrader()
+	u.OnMessage(func(c *websocket.Conn, messageType websocket.MessageType, data []byte) {
+		fmt.Printf("received: %s\n",string(data))
+		connectedClients <- c
+	})
+
+	u.OnClose(func(c *websocket.Conn, err error) {
+	})
+
+	return u
+}
+
+func main() {
+	flag.Parse()
+	engine := nbhttp.NewEngine(nbhttp.Config{})
+	err := engine.Start()
+	if err != nil {
+		fmt.Printf("nbio.Start failed: %v\n", err)
+		return
+	}
+
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
+	}
+
+	connectedClients = make(chan *websocket.Conn, *clients)
+	for i := 0; i < *clients; i++ {
+		u := url.URL{Scheme: "wss", Host: "localhost:8888", Path: "/wss"}
+		dialer := &websocket.Dialer{
+			Engine:          engine,
+			Upgrader:        newUpgrader(),
+			DialTimeout:     time.Second * 3,
+			TLSClientConfig: tlsConfig,
+		}
+		c, _, err := dialer.Dial(u.String(), nil)
+		if err != nil {
+			panic(fmt.Errorf("dial: %v", err))
+		}
+		connectedClients <- c
+	}
+
+	go func() {
+			c := <-connectedClients
```

Comment:
2. you only get client one time here.

Reply:
5. this will be fine:
```golang
	go func() {
		i := 0
		for c := range connectedClients {
			i++
			c.WriteMessage(websocket.TextMessage, []byte(fmt.Sprintf("message %d", i)))
		}
	}()
```